### ISSUE TITLE
license tool should categorise all third-party deps in the licenses_third_party category

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#136390]
## TL;DR
As we move third-party packages sync location from src/third_party (under the buildroot directory) to src/flutter/third_party (under the engine directory), performance of the license aggregation tool will get worse for those working on Flutter engine code.

Importantly, this is entirely a caching/performance issue, the output of the tool is unaffected.

## Background
The licence tool includes a performance optimisation that separates our licences into three categories:
* Flutter licences
* Third-party licences
* Fuchsia licences

This performance optimisation _does not_ affect the licences eventually emitted to our licence file. What it _does_ do is use change detection and caching to recompute the licence file _only_ for code that changed, and pull the other categories from the cached copy. In this way, changes to checked-in code in the engine only re-trigger Flutter licence aggregation, and not the much-slower third-party licence aggregation, where no changes occurred. Changes to third-party code trigger the slower third-party aggregation, but not (for example) Fuchsia aggregation.

Today, flutter/third_party actually falls under Flutter licenses rather than third-party licences. This has been non-problematic because historically that code consisted of Flutter-authored code like libtxt that was authored by us, but independent of the rest of the system, as well as forks of third-party code like third_party/accessibility, which are maintained by the Flutter team. Neither was subject to significant churn and neither added much run time to that licenses_flutter category.

## Problem description
As we progress on merging the buildroot to the engine and moving third-party dependencies (synced via the DEPS file) to pull to src/flutter/third_party rather than src/third_party, this code will increase the set of code categorised into the licenses_flutter category, and decrease (and eventually eliminate) code categorised into the licences_third_party category, thus stepwise undoing the current performance optimisation. Again, this _does not_ affect the final set of licences, but will significantly increase script run time for anyone changing engine code.

To maintain the current performance gains I propose that we categorise code under src/flutter/third_party into the licenses_third_party category. If we find that we're actively working on any of the code checked into that directory and CI time is problematic, we could add an exceptions list that moves the affected code back into the faster-to-compute licenses_flutter bin.

**Comment by cbracken** [comment]
Breadcrumb for where we break out the categories:
https://github.com/flutter/engine/blob/4b02631b59bf071e967f7bfc4c6a949b43c51e14/tools/licenses/lib/main.dart#L2098-L2111

**Comment by cbracken** [comment]
Quick summary: right now we just break out licence "bins" by the top-level directory under the buildroot (`src`). That was a reasonable(-ish) strategy when we instituted it (at the time, flutter/third_party was really just libtxt), and will be a reasonable strategy once the buildroot is gone (flutter/third_party will be the full third_party).

Ideally, what I think we want during the migration is the ability to selectively exclude directories from one bin and put them in another. Maybe just make the current implicit mapping of top-level directory -> bin a little more explicit by pulling out a function, and supporting the idea of overrides for specific paths. Something like an overrides list where we just say `flutter/third_party` -> `third_party_licences`, then add exclusions/additions optional named params to the licence crawl function.

**Comment by cbracken** [comment]
With the repo merge on the way, I've unsassigned myself. Once everything is landed, someone (maybe me) will likely want to pick this up again.
